django/contrib/admin/sites.py

```diff
@@ -245,20 +247,23 @@ def wrapper(*args, **kwargs):
 
         # Add in each model's views, and create a list of valid URLS for the
         # app_index
-        valid_app_labels = []
+        valid = defaultdict(set)
```

Comment:
yes, that isn't the cleanest way. There is a security whole as well. You could access an unregistered objects in with the same name in a different app. I fixed it.